src/services/user.service.js

```diff
@@ -0,0 +1,44 @@
+const { data } = require('../data');
+
+const getAll = () => {
+  return data.users;
+};
+
+const getById = (id) => {
+  return data.users.find((user) => user.id === id) || null;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getById(id);
+
+  Object.assign(user, { name });
+
+  return user;
```

Comment:
In the `updateUser` function, you are directly modifying the user object that is stored in the `data` array. This could lead to unintended side effects in other parts of the code that are using the same user object. Consider creating a new object with the updated values instead of modifying the existing one. Also, you should handle the case when the user is not found.